### PR TITLE
Update the patch releases calendar with February patch releases

### PR DESCRIPTION
**Pull request by xmudrii**
- Add February patch releases to the patch releases calendar
- Remove 1.23 from the patch releases calendar and mark it as EOL
- Add tentative release dates for May and June

/assign @saschagrunert @cpanato @puerco @Verolop 
cc @kubernetes/release-engineering

### CODE REVIEW

File: content/en/releases/patch-releases.md [hunk 1]
```diff
@@ -78,9 +78,10 @@ releases may also occur in between these.
 
 | Monthly Patch Release | Cherry Pick Deadline | Target date |
 | --------------------- | -------------------- | ----------- |
-| February 2023         | 2023-02-10           | 2023-02-15  |
 | March 2023            | 2023-03-10           | 2023-03-15  |
 | April 2023            | 2023-04-07           | 2023-04-12  |
+| May 2023              | 2023-05-12           | 2023-05-17  |
```
Comment (xmudrii) [inline content/en/releases/patch-releases.md:83]: There's more than one month between April and May patch releases. We could release May patch releases one week earlier, but:

- there'll be low activity week prior and after KubeCon
- there are public holidays around 1st May in some countries

Should we leave it for May 17 or move it to May 10?

File: data/releases/schedule.yaml [hunk 2]
```diff
@@ -4,10 +4,15 @@ schedules:
...
-    release: 1.26.2
-    cherryPickDeadline: 2023-02-10
-    targetDate: 2023-02-15
+    release: 1.26.3
+    cherryPickDeadline: 2023-03-10
+    targetDate: 2023-03-15
   previousPatches:
+    - release: 1.26.2
+      cherryPickDeadline: 2023-02-10
+      targetDate: 2023-02-15
```
Comment (xmudrii) [inline data/releases/schedule.yaml:13]: Should this that be the original target date or when we actually did the release (February 28)?

File: content/en/releases/patch-releases.md (hunk 1 above)
Comment (saschagrunert) [reply content/en/releases/patch-releases.md:83]: I think going for May 17 is totally fine :+1:

File: data/releases/schedule.yaml (hunk 2 above)
Comment (saschagrunert) [reply data/releases/schedule.yaml:13]: When we actually released.

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**